src/mlpack/methods/ann/rnn.hpp

```diff
@@ -281,22 +282,22 @@ class RNN
   std::vector<LayerTypes> network;
 
   //! The matrix of data points (predictors).
-  arma::mat predictors;
+  MatrixType predictors;
 
   //! The matrix of responses to the input data points.
-  arma::mat responses;
+  MatrixType responses;
 
   //! Matrix of (trained) parameters.
-  arma::mat parameter;
+  MatrixType parameter;
```

Comment:
Hi @etano, thanks again for taking the time to do this.  I think that if we templatize the internal parameters like this, we can end up with a bad situation: the user may want to pass `arma::sp_mat` (a sparse matrix) as their input, but we don't want to store a sparse matrix internally for the parameters (because they will generally be non-sparse).

One option is that we can do something like this:

```
typedef MatrixType::elem_type ElemType;

arma::Mat<ElemType> parameter;
```

and this could sidestep the problem.  What do you think?

(I'd also suggest changing to `MatType` like @zoq suggested.)